Jenkinsfile

```
@@ -507,12 +507,16 @@ def target(Map args = [:]) {
 }
 
 /**
-* This method wraps the node call with some latency to avoid the known issue with the scalabitity in gobld.
+* This method wraps the node call for two reasons:
+*  1. with some latency to avoid the known issue with the scalabitity in gobld.
+*  2. allocate a new workspace to workaround the flakiness of windows workers with deleteDir
 */
 def withNode(String label, Closure body) {
   sleep randomNumber(min: 10, max: 200)
   node(label) {
-    body()
+    ws(env.BUILD_TAG) {
```

Unfortunately it works half way:

```

drwxr-xr-x  5 jenkins jenkins 4096 Mar  2 11:34 workspace
drwxr-xr-x 17 jenkins jenkins 4096 Mar  2 11:46 .
drwxr-xr-x  3 jenkins jenkins 4096 Mar  2 11:57 jenkins-Beats-beats-PR-24289-1
drwxr-xr-x  2 jenkins jenkins 4096 Mar  2 12:00 jenkins-Beats-beats-PR-24289-1@tmp
```

It's required to add the `workspace` folder :/ , and probably it's worthy to use the `"${JOB_BASE_NAME}-${BUILD_NUMBER}"` to avoid the well known dir size in windows.

```
BUILD_NUMBER
The current build number, such as "153"

JOB_BASE_NAME
Short Name of the project of this build stripping off folder paths, such as "foo" for "bar/foo".

BUILD_TAG
String of "jenkins-${JOB_NAME}-${BUILD_NUMBER}". All forward slashes ("/") in the JOB_NAME are replaced with dashes ("-"). Convenient to put into a resource file, a jar file, etc for easier identification.
```

```suggestion
    ws("workspace/${JOB_BASE_NAME}-${BUILD_NUMBER}") {
```